omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_8/PersonResource1_8.java

```diff
@@ -131,6 +131,7 @@ public DelegatingResourceDescription getUpdatableProperties() throws ResourceDoe
 		description.addProperty("gender");
 		description.addProperty("birthdate");
 		description.addProperty("birthdateEstimated");
+		description.addProperty("birthtime");
```

Comment:
The OpenMRS platform versions supported by this resource do not have the `birthtime` property.

Reply:
But when we register new patient in bahmni-standard by entering the birthtime and click save and then if i do some changes in the registration form and click again save/start visit, it's giving error: "org.openmrs.module.webservices.rest.web.response.ConversionException: Some properties are not allowed to be set: birthtime".

Reply:
Which openmrs platform version are you running?

Reply:
Version: 2.5.12 Build 406cac

Reply:
This resource supports up to openmrs platform version `1.10.3`. So when one runs it on any platform version from `1.10.3` and below, they will get an error because this property does not exist in those versions.

Reply:
But i am running on version 2.5.12 still getting error "birthtime" property not allowed to set. that'shy i a make those changes and raise pr.

Reply:
This was added here: https://openmrs.atlassian.net/browse/RESTWS-730
Which version of the restwebservices module are you running?

Reply:
2.39.0

Reply:
Strange! Can you share the full stack trace via pastebin.com?

Reply:
inside pom.xml of reswebservices it's showing <version>2.42.0-SNAPSHOT</version> but in openmrs-distro-bahmni the version is <webServicesRestVersion>2.39.0</webServicesRestVersion>

Reply:
Version  2.42.0-SNAPSHOT already has this property. Are you able to reproduce this problem at? https://demo.openmrs.org/

Reply:
when i try to login i got invalid username. what location i have to select for this session?

Reply:
Choose any location. UserName: `admin` Password: `Admin123`

Reply:
when i register the patient, it is not asking for birth-time, asking only birth day,month and year in this https://demo.openmrs.org/

Reply:
i think that's why it is not reproducing this birthtime property problem

Reply:
Just make a REST call where the person details include birthtime

Reply:
I have no idea, can you please tell how to do that. i login on this link https://demo.openmrs.org/ and try to register the patient. so its not giving any error and during regisration it is not asking for birthtime.

Reply:
You can use a tool like postman and make a REST call for patient registration where you include birthtime. You can even use this: https://demo.openmrs.org/openmrs/module/webservices/rest/test.htm

Reply:
i got this

Reply:
Change the URI such that you make the REST call to: https://demo.openmrs.org/

Reply:
i have pass this uri: /openmrs/ws/rest/v1/patient/808ea54d-71e4-4d27-b418-7c196e3b390c. still get the same response

Reply:
this is my body-content https://pastebin.com/N1C1NU2P

Reply:
Do a simple body for  https://demo.openmrs.org/

Reply:
Will my above body will not work?

Reply:
it is not giving error when we create the patient, once we create patient in bahmni-standard and click on save button and again update some changes and then again click on save button it's giving error "birthtime property is not allowe with set".

Reply:
It would be great if you can create a test to reproduce that error.

Reply:
@dkayiwa The failing scenario is when we try to update a person's birthtime. Please find the unit test below

```	
@Test
	public void shouldEditAPersonBirthTime() throws Exception {
		Person person = service.getPersonByUuid(getUuid());
		String json = "{\"birthtime\": \"1970-01-01T20:20:00.000\"}";
		SimpleObject response = deserialize(handle(newPostRequest(getURI() + "/" + getUuid(), json)));
		assertNotNull(response);
		Object responsePersonContents = PropertyUtils.getProperty(response, "person");
		assertNotNull(responsePersonContents);
		//assertEquals("1970-01-01T20:20:00.000+0000", PropertyUtils.getProperty(responsePersonContents, "birthtime").toString());
	}
```

Error logs for the failing test : https://pastebin.com/pqJviEZZ

Reply:
Override `getUpdatableProperties` in `PersonResource1_10` and add `birthtime`

Reply:
still above test case giving error in my locale but during registration in bahmni-registration when i update the birthtime it's not giving error. how?

Reply:
Is there any reason why you have not committed the test together with your addition of the birthtime property?

Reply:
I commit the test now, actually, I forgot to do that.

Reply:
sorry for that